File: ANTWebAPI/src/ANTWebAPI.PL/Mappers/ArticleMapper.cs
using ANTWebAPI.BLL.Models;
using ANTWebAPI.PL.DTOs;

namespace ANTWebAPI.PL.mappers;

/*
 * Article mapper is used for extension methods to map ArticleDTO in the Article model and vice versa
 */
public static class ArticleMapper
{
    /*
     * ToDto is extension method which returns ArticleDTO from Article
     * 
     * @param article
     * 
     * @return ArticleDTO
     */
    public static ArticleDTO ToDto(this Article article) => new()
    {
        Id = article.Id,
        Title = article.Title,
        Catalog = article.Catalog.ToDto(),
        Description = article.Description,
        DateOrBanner = article.DateOrBanner
    };
    /*
     * ToModel is extension method which returns Article from ArticleDTO
     * 
     * @param articleDto
     * 
     * @return Article
     */
    public static Article ToModel(this ArticleDTO articleDto) => new()
    {
        Id = articleDto.Id,
        Title = articleDto.Title,
        CatalogId = articleDto.Catalog.Id,
        Description = articleDto.Description,
        DateOrBanner = articleDto.DateOrBanner
    };
}

File: ANT-Android-App/data/src/main/java/com/fredprojects/ant/data/mappers/ArticleMapper.kt
package com.fredprojects.ant.data.mappers

import com.fredprojects.ant.data.local.ArticleEntity
import com.fredprojects.ant.data.remote.dto.ArticleDto
import com.fredprojects.ant.domain.models.Article
import kotlinx.serialization.json.Json

/**
 *  Mapper for mapping ArticleDto to Article and vice versa
 *  @return Article
 */
fun ArticleDto.toModel() = Article(
    title = title,
    description = description,
    date = dateOrBanner,
    articleType = catalog.name,
    content = content,
    id = id
)
/**
 *  Mapper for mapping ArticleDto to ArticleEntity and vice versa
 *  @return ArticleEntity
 */
fun ArticleDto.toEntity(catalogId: Int, pageNumber: Int) = ArticleEntity(
    id = id,
    title = title,
    description = description,
    date = dateOrBanner,
    articleType = catalog.name,
    pageNumber = pageNumber.toLong(),
    catalogId = catalogId.toLong(),
    content = Json.encodeToString(content)
)
/**
 *  Mapper for mapping ArticleEntity to Article and vice versa
 *  @return Article
 */
fun ArticleEntity.toModel() = Article(
    title = title,
    description = description,
    date = date,
    articleType = articleType,
    content = Json.decodeFromString<List<String>>(content),
    id = id
)